components/function-controller/internal/webhook/conversion_webhook.go

```diff
@@ -282,16 +299,20 @@ func (w *ConvertingWebhook) convertSpecV1Alpha2ToV1Alpha1(in *serverlessv1alpha2
 	out.Spec.Env = in.Spec.Env
 	out.Spec.Runtime = serverlessv1alpha1.Runtime(in.Spec.Runtime)
 	out.Spec.RuntimeImageOverride = in.Spec.RuntimeImageOverride
+
 	if in.Spec.ScaleConfig != nil {
 		out.Spec.MaxReplicas = in.Spec.ScaleConfig.MaxReplicas
 		out.Spec.MinReplicas = in.Spec.ScaleConfig.MinReplicas
 	}
-
-	out.Spec.BuildResources = in.Spec.ResourceConfiguration.Build.Resources
-	out.Spec.Resources = in.Spec.ResourceConfiguration.Function.Resources
-
-	out.Spec.Labels = in.Spec.Template.Labels
-
+	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Build != nil && in.Spec.ResourceConfiguration.Build.Resources != nil {
+		out.Spec.BuildResources = *in.Spec.ResourceConfiguration.Build.Resources
+	}
+	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Function != nil && in.Spec.ResourceConfiguration.Function.Resources != nil {
+		out.Spec.Resources = *in.Spec.ResourceConfiguration.Function.Resources
+	}
```

Comment:
Put it in `convertResourcesV1Alpha2ToV1Alpha1` method to improve readibility.

Reply:
done